tensorboard/backend/event_processing/data_provider.py

```diff
@@ -125,3 +129,151 @@ def _convert_scalar_event(self, event):
+  ):
+    del run_tag_filter # ignored for now
+
+    # TODO(davidsoergel, wchargin): consider images, etc.
+    # Note this plugin_name can really just be 'graphs' for now; the
+    # v2 cases are not handled yet.
+    if plugin_name != graphs_metadata.PLUGIN_NAME:
+      logger.warn("Directory has no blob data for plugin %r", plugin_name)
+      return {}
+
+    result = collections.defaultdict(
+      lambda: collections.defaultdict(lambda: []))
+    for (run, run_info) in six.iteritems(self._multiplexer.Runs()):
+      tag = None
+      if not run_info['graph']:
+        continue
+
+      time_series = result[run][tag]
+
+      wall_time = 0  # dummy value for graph
+      step = 0  # dummy value for graph
+      index = 0 # dummy value for graph
+
+      # In some situations these blobs may have directly accessible URLs.
+      # But, for now, we assume they don't.
+      graph_url = None
+      graph_blob_key = _encode_blob_key(
+          experiment_id, plugin_name, run, tag, step, index)
+      blob_ref = provider.BlobReference(graph_blob_key, graph_url)
+
+      datum = provider.BlobSequenceDatum(
+          wall_time=wall_time,
+          step=step,
+          values=(blob_ref,),
+      )
+      time_series.append(datum)
+    return result
+
+  def read_blob(self, blob_key):
+    # note: ignoring nearly all key elements: there is only one graph per run.
+    (unused_experiment_id, plugin_name, run, unused_tag, unused_step,
+        unused_index) = _decode_blob_key(blob_key)
+
+    # TODO(davidsoergel, wchargin): consider images, etc.
+    if plugin_name != graphs_metadata.PLUGIN_NAME:
+      logger.warn("Directory has no blob data for plugin %r", plugin_name)
+      return None
+
+    serialized_graph = self._multiplexer.SerializedGraph(run)
+
+    # TODO(davidsoergel): graph_defs have no step attribute so we don't filter
+    # on it.  Other blob types might, though.
+
+    if not serialized_graph:
+      logger.warn("No blob found for key %r", blob_key)
+      return None
+
+    # TODO(davidsoergel): consider internal structure of non-graphdef blobs.
+    # In particular, note we ignore the requested index, since it's always 0.
+    return serialized_graph
+
+# TODO(davidsoergel): deduplicate with other implementations
+def _encode_blob_key(experiment_id, plugin_name, run, tag, step, index):
+  """Generate a blob key: a short, URL-safe string identifying a blob.
+
+  A blob can be located using a set of integer and string fields; here we
+  serialize these to allow passing the data through a URL.  Specifically, we
+  1) construct a tuple of the arguments in order; 2) represent that as an
+  ascii-encoded JSON string (without whitespace); and 3) take the URL-safe
+  base64 encoding of that, with no padding.  For example:
+
+      1)  Tuple: (123, "graphs", "train", "graph_def", 2, 0)
+      2)   JSON: "[123,"graphs","train","graph_def",2,0]"
+      3) base64: WzEyMywiZ3JhcGhzIiwidHJhaW4iLCJncmFwaF9kZWYiLDIsMF0
```

Comment:
Experiment IDs are like `"123"`, not `123`, no?
      
```
      1)  Tuple: ("123", "graphs", "train", "graph_def", 2, 0)
      2)   JSON: "["123","graphs","train","graph_def",2,0]"
      3) base64: IjEyMyIsImdyYXBocyIsInRyYWluIiwiZ3JhcGhfZGVmIiwyLDA
```

(Thanks for including this example—it’s quite helpful.)